sparkle/src/components/Table.tsx

```diff
@@ -0,0 +1,296 @@
+import {
+  type ColumnDef,
+  ColumnFiltersState,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  type SortingState,
+  useReactTable,
+} from "@tanstack/react-table";
+import React, { ReactNode, useEffect, useState } from "react";
+
+import { Avatar, MoreIcon } from "@sparkle/index";
+import { ArrowDownIcon, ArrowUpIcon } from "@sparkle/index";
+import { classNames } from "@sparkle/lib/utils";
+
+import { Icon } from "./Icon";
+
+type FirstColumnWidth = "expanded" | "normal";
+
+interface TableProps<TData, TValue> {
+  data: TData[];
+  columns: ColumnDef<TData, TValue>[];
+  className?: string;
+  width?: FirstColumnWidth;
```

Comment:
Let's have this behavior by default? Or let the user overrides it when defining the table's columns.